packages/react-instantsearch/src/connectors/connectScrollTo.js

```diff
@@ -32,6 +32,27 @@ export default createConnector({
       null,
       currentRefinement => currentRefinement
     );
-    return { value };
+
+    /*
+      We need to keep track of the search state to know if there's a change in the app
+      that was not triggered by the props.scrollOn (id) or the Configure widget. 
+      This is useful when using ScrollTo in combination of Pagination. As pagination
+      can be change by every widget, we want to scroll only if it cames from the pagination
+      widget itself. We also remove the configure key to do this comparaison because for 
+      now configure values are not present in the search state before a first refinements and will false 
+      the results. 
+      See: https://github.com/algolia/react-instantsearch/issues/164
+    */
+    const cleanedSearchState = cleanUpValue(
+      cleanUpValue(searchState, this.context, 'configure'),
+      this.context,
+      id
+    );
+
+    const hasNotChanged = isEqual(this._prevSearchState, cleanedSearchState);
```

Comment:
So it's called twice each time your perform a refinement. (before and after the query is resolved). It's not called when scrolling. 

The isEqual is done on the `searchState` of RIS, not the helper one. 

However I believe we can use the `shallowEqual` here, wdyt?